include/sensors/core/mmr920C04.hpp

```diff
@@ -0,0 +1,281 @@
+#include "sensors/core/mmr920C04_registers.hpp"
+#include "sensors/core/sensors.hpp"
+
+namespace mmr920C04 {
+
+using namespace can_ids;
+using namespace mmr920C04_registers;
+using namespace sensors_registers;
+
+template <class I2CQueueWriter, message_writer_task::TaskClient CanClient>
+class MMR92C04 {
+  public:
+    MMR92C04(I2CQueueWriter &writer, CanClient &can_client)
+        : writer(writer), can_client(can_client) {}
+
+    /**
+     * @brief Check if the MMR92C04 has been initialized.
+     * @return true if the config registers have been written at least once,
+     * false otherwise.
+     */
+    [[nodiscard]] auto initialized() const -> bool { return _initialized; }
+
+    auto register_map() -> MMR920C04RegisterMap & { return _registers; }
+
+    auto get_sensor_id() -> SensorType { return SensorType::pressure; }
+
+    auto get_host_id() -> NodeId { return NodeId::host; }
+
+    auto get_threshold() -> int32_t { return threshold; }
+
+    auto set_threshold(int32_t new_threshold) -> void {
+        threshold = new_threshold;
+    }
+
+    auto write(Registers reg, uint32_t command_data) -> void {
+        writer.write(command_data, ADDRESS, static_cast<uint8_t>(reg));
+    }
+
+    auto read(Registers reg) -> void {
+        writer.read(
+            ADDRESS, [this, reg]() { send_to_can(this, reg); },
+            [this, reg](auto message_a) { handle_data(message_a, this, reg); },
+            static_cast<uint8_t>(reg));
+    }
+
+    auto poll_read(Registers reg, uint16_t number_reads) -> void {
+        writer.single_register_poll(
+            ADDRESS, number_reads, DELAY,
+            [this, reg]() { send_to_can(this, reg); },
+            [this, reg](auto message_a) { handle_data(message_a, this, reg); },
+            static_cast<uint8_t>(reg));
+    }
+
+    auto write_config() -> bool {
+        if (!reset(_registers.reset)) {
+            return false;
+        }
+        if (!set_measure_mode4(_registers.measure_mode_4)) {
+            return false;
+        }
+        _initialized = true;
+        return true;
+    }
+
+    auto set_measure_mode1(MeasureMode1 reg) -> bool {
+        if (set_register(reg)) {
+            _registers.measure_mode_1 = reg;
+            return true;
+        }
+        return false;
+    }
+
+    auto set_measure_mode2(MeasureMode2 reg) -> bool {
+        if (set_register(reg)) {
+            _registers.measure_mode_2 = reg;
+            return true;
+        }
+        return false;
+    }
+
+    auto set_measure_mode3(MeasureMode3 reg) -> bool {
+        if (set_register(reg)) {
+            _registers.measure_mode_3 = reg;
+            return true;
+        }
+        return false;
+    }
+
+    auto set_measure_mode4(MeasureMode4 reg) -> bool {
+        if (set_register(reg)) {
+            _registers.measure_mode_4 = reg;
+            return true;
+        }
+        return false;
+    }
+
+    auto get_pressure(bool poll = false, uint16_t sample_rate = 0) -> void {
```

Comment:
These three get_pressure methods look about the same. Can there be a function that takes the register as input and reduces the duplication of logic?